promql/promqltest/README.md

```diff
@@ -83,58 +85,82 @@ The syntax is as follows:
+    * `no_info` expects the query to return no info annotation.
+    * `no_warn` expects the query to return no warn annotation.
+    * `ordered` expects the query to return the results in the specified order.
+* `<match_type>` (optional) specifies message matching type for annotations:
+    * `msg` for exact string match.
+    * `regex` for regular expression match.
+    * **Not applicable** for `ordered`, `no_info`, and `no_warn`.
+* `<string>` is the expected annotation message.
+
 For example:
 
 ```
 eval instant at 1m sum by (env) (my_metric)
+    expect warn
+    expect no_info
     {env="prod"} 5
     {env="test"} 20
     
 eval range from 0 to 3m step 1m sum by (env) (my_metric)
+    expect warn msg something went wrong
+    expect info regex something went (wrong|boom)
     {env="prod"} 2 5 10 20
     {env="test"} 10 20 30 45
-```
 
-To assert that a query succeeds with an info or warn annotation, use the
-`eval_info` or `eval_warn` commands, respectively.
+eval instant at 1m ceil({__name__=~'testmetric1|testmetric2'})
+expect fail
 
-Instant queries also support asserting that the series are returned in exactly
-the order specified: use `eval_ordered instant ...` instead of `eval instant
-...`. `eval_ordered` ignores any annotations. The assertion always fails for
-matrix results.
+eval instant at 1m ceil({__name__=~'testmetric1|testmetric2'})
+expect fail msg "vector cannot contain metrics with the same labelset"
 
-To assert that a query fails, use the `eval_fail` command. `eval_fail` does not
-expect any result lines. Instead, it optionally accepts an expected error
-message string or regular expression to assert that the error message is as
-expected.
-
-For example:
+eval instant at 1m ceil({__name__=~'testmetric1|testmetric2'})
+expect fail regex "vector cannot contain metrics .*|something else went wrong"
 
+eval instant at 1m sum by (env) (my_metric)
+expect ordered
+{env="prod"} 5
+{env="test"} 20
 ```
-# Assert that the query fails for any reason without asserting on the error message.
-eval_fail instant at 1m ceil({__name__=~'testmetric1|testmetric2'})
 
-# Assert that the query fails with exactly the provided error message string.
-eval_fail instant at 1m ceil({__name__=~'testmetric1|testmetric2'})
-    expected_fail_message vector cannot contain metrics with the same labelset
+There can be multiple `<expect>` lines for a given `<type>`. Each `<type>` validates its corresponding annotation, error, or ordering while ignoring others.
 
-# Assert that the query fails with an error message matching the regexp provided.
-eval_fail instant at 1m ceil({__name__=~'testmetric1|testmetric2'})
-    expected_fail_regexp (vector cannot contain metrics .*|something else went wrong)
-```
+Every `<expect>` line must match at least one corresponding annotation or error.
+
+If at least one `<expect>` line of a specific `<type>` is present, then all corresponding annotations must have a matching `expect` line.
```

Comment:
Maybe clarify that this is only for warns and infos.

```suggestion
If at least one `<expect>` line of type `warn` or `info` is present, then all corresponding annotations must have a matching `expect` line.
```

Reply:
Done!